qiskit/pulse/instructions/instruction.py

```diff
@@ -197,16 +201,25 @@ def append(self, schedule, name: Optional[str] = None):
     @property
     def parameters(self) -> Set:
         """Parameters which determine the instruction behavior."""
+
+        def _get_parameters_recursive(obj):
+            params = set()
+            if hasattr(obj, "parameters"):
+                for param in obj.parameters:
+                    if isinstance(param, Parameter):
+                        params.add(param)
+                    else:
+                        param = param | _get_parameters_recursive(param)
```

Comment:
did you mean `params |= _get_parameters_recursive(param)`? (note the different assignment)

Reply:
I think you are right and this probably means there needs to be another test added that uses a `ParameterExpression`.

Reply:
Thanks, updated in https://github.com/Qiskit/qiskit-terra/pull/8005/commits/329c6bc386e81c8a73f6da60e64d3bc47a492f9e